DataMgr/ForeignStorage/ArrowCsvForeignStorage.cpp

```diff
@@ -32,24 +32,43 @@ void ArrowCsvForeignStorage::read(const ChunkKey& chunk_key,
   std::array<int, 3> col_key {chunk_key[0], chunk_key[1], chunk_key[2]};
   auto &frag = g_columns.at(col_key).at(chunk_key[3]);
   int64_t sz, copied = 0;
+  std::shared_ptr<arrow::ArrayData> prev_data = nullptr;
```

Comment:
operating shared pointers is expensive excessive op when no ownership guarantee is required